main/main.c

```diff
@@ -659,7 +659,40 @@ PHPAPI ZEND_COLD void php_log_err_with_severity(char *log_message, int syslog_ty
 	if (PG(error_log) != NULL) {
 #ifdef HAVE_SYSLOG_H
 		if (!strcmp(PG(error_log), "syslog")) {
-			php_syslog(syslog_type_int, "%s", log_message);
+			static char nvt[] = " !\"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~";
```

Comment:
> Most of encodings have lower 128 bytes compatible with ASCII, so replacing invalid bytes would likely not break anything, maybe some very exotic encodings.

Are we talking about `iso-8859-n` or UTF?  Because when you start having multi-byte encodings the range 0x00-0x2f is definitely going to be used, including 0x1b (ESC) of course.

> I would suggest to go by the full RFC compliance, as the relevant part is being touched anyway.

Well, commits 2 and 3 try to be fully RFC compliant, to the extent that the RFC is explicit.  There are some places where there's a bit of hand-waving.

I'm not familiar with internals@.